File: 0217_Problem_1.rs
// code_report Solution
// Problem Link (Contest):  https://leetcode.com/contest/weekly-contest-217/problems/richest-customer-wealth/
// Problem Link (Practice): https://leetcode.com/problems/richest-customer-wealth/

pub fn maximum_wealth(accounts: Vec<Vec<i32>>) -> i32 {
    accounts.iter()
            .map(|x| x.iter().sum())
            .max()
            .unwrap()
}

File: 0217_Problem_1.scala
// code_report Solution
// Problem Link (Contest):  https://leetcode.com/contest/weekly-contest-217/problems/richest-customer-wealth/
// Problem Link (Practice): https://leetcode.com/problems/richest-customer-wealth/

def maximumWealth(accounts: Array[Array[Int]]): Int = {
    accounts.map(_.sum).max
}

// Alternative (remove braces)
def maximumWealth(accounts: Array[Array[Int]]): Int =
    accounts.map(_.sum).max
